### PR TITLE
[script.service.playbackresumer] 1.2.0

### PR DESCRIPTION
**Pull request by bossanova808**
### Description

- Taking over an abandoned addon as @bradvido88 is absent (see https://forum.kodi.tv/showthread.php?tid=183084)
- Updated for Leia (should work all the way back to Gotham though)
- (Version number is 1.2.0 rather than 1.0 - so folks with this manually installed will get the update)

### Checklist:
- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] v1.0.0

### CODE REVIEW

File: script.service.playbackresumer/addon.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    <summary lang="en">Periodically sets the resume point of videos and can automatically resume last played video if XBMC crashes.</summary>
+    <description lang="en">
+        This addon runs in the background as a service and will periodically update the resume point while videos are playing. It will also automatically resume a video if XBMC was shutdown while playing it. See setting to configure how often the resume point is set and whether to automatically resume. 
+    </description>
+    <platform>all</platform>
+	<language></language> <!-- *content* language, not applicable to this service -->
+	<license>GNU GENERAL PUBLIC LICENSE Version 3, 29 June 2007</license>
+    <website>https://github.com/bossanova808/script.service.playbackresumer</website>
+    <source>https://github.com/bossanova808/script.service.playbackresumer</source>
+    <forum>http://forum.xbmc.org/showthread.php?tid=183084</forum>
```
Comment (Rechi) [inline script.service.playbackresumer/addon.xml]:
```suggestion
    <forum>https://forum.kodi.tv/showthread.php?tid=183084</forum>
```

File: script.service.playbackresumer/default.py [hunk 2]
```diff
@@ -0,0 +1,466 @@
...
+    xbmc.log("### [%s] - %s" % (__scriptname__,msg,),level=xbmc.LOGDEBUG )
+
+
+# Helper function to get string type from settings
+def getSetting(setting):
+    return __addon__.getSetting(setting).strip()
+
+# Helper function to get bool type from settings
+def getSettingAsBool(setting):
+    return getSetting(setting).lower() == "true"
```
Comment (Rechi) [inline script.service.playbackresumer/default.py]:
```suggestion
    return __addon__.getSettingBool(setting)
```

File: script.service.playbackresumer/default.py [hunk 3]
```diff
@@ -0,0 +1,466 @@
...
+
+# Load the addon setting into our globals
+
+def loadSettings():
+
+    global saveintervalsecs
+    global resumeonstartup
+    global autoplayrandom
+
+    saveintervalsecs = int(float(__addon__.getSetting("saveintervalsecs")))
```
Comment (Rechi) [inline script.service.playbackresumer/default.py]:
```suggestion
    saveintervalsecs = __addon__.getSettingInt("saveintervalsecs")
```

File: script.service.playbackresumer/default.py [hunk 4]
```diff
@@ -0,0 +1,466 @@
...
+    def onPlayBackSeekChapter( self, chapter ):
+        log("Playback seeked (chapter)")
+        updateResumePoint(xbmc.Player().getTime())
+
+    # These two events have changed from Kodi Leia - in general, post Leia, we want to wait onAVStarted.  
+    # Before Leia we use onPlayBackStarted.
+    
+    def onPlayBackStarted( self ):
+        if __kodiversion__ < 17.9:
+            handlePlayback()
```
Comment (Rechi) [inline script.service.playbackresumer/default.py]: no need to keep this, as minimum Kodi version supported by this add-on is Leia

File: script.service.playbackresumer/default.py [hunk 5]
```diff
@@ -0,0 +1,466 @@
...
+
+    def onSettingsChanged( self ):
+        loadSettings()
+
+    def onAbortRequested(self):
+        log("Abort Requested!")
+
+# MAIN - get this thing going...
+
+log( "[%s] - Version: %s Started" % (__scriptname__,__version__))
```
Comment (Rechi) [inline script.service.playbackresumer/default.py]: Kodi logs that already

File: script.service.playbackresumer/README.md [hunk 6]
```diff
@@ -0,0 +1,12 @@
+# script.service.playbackresumer
+
+A simple addon to periodically update the resume point of the currently playing video automatically in the background, so that playback progess is not lost in the event of a Kodi crash.
+
+Optionally atuomatically resume playback automatically when xbmc is restarted after a crash, or trigger playback of a random video.
+
+Original code by bradvido88 recovered from the main Kodi forum thread on this, and from May 2019 updated & maintained in this repo by bossanova808.
+
+Main Kodi forum thread here:
+http://forum.kodi.tv/showthread.php?tid=183084
```
Comment (Rechi) [inline script.service.playbackresumer/README.md]:
```suggestion
https://forum.kodi.tv/showthread.php?tid=183084
```

**Comment by bossanova808** [comment]
Thanks for the review...was just, as I originally thought it was in the Gotham repo like most other things.

I have a bunch of these addons to convert to Python3 it seems!  Will have to find some time for that...I'll be back!

**Pull request closed without merging**